packages/react-native-editor/__device-tests__/helpers/utils.js

```diff
@@ -140,6 +142,13 @@ const setupDriver = async () => {
 			}
 
 			desiredCaps.app = path.resolve( localIOSAppPath );
+
+			if ( isCI ) {
```

Comment:
I think having another code path just for local CI could further complicate things. WDYT about adding `test:e2e:build-wda` in `test:e2e:ios:local` as an additional step and removing the check for CI so both local and CI runs stay the same?

Reply:
The difference between local and CI regarding the command `test:e2e:build-wda` is that for CI the WDA cache gets invalidated with the build cache key so if there's any changes in the native side, the WDA is rebuilt but this won't happen for local. This could produce an error if you change for example the Xcode version without deleting the derived data folder `ios/build/WDA`.

For local, my idea was to let Appium decide when the WDA should be rebuilt, that's why we have another code path but I'd be happy to figure out a different approach.

Reply:
When we put `test:e2e:build-wda` inside `test:e2e:ios:local` it should also always rebuild WDA when ran locally, right? We can add a `rm -rf ios/build/WDA` in `test:e2e:build-wda` to be safe, but it seems to me that it's the same case for building with `test:e2e:build-app:ios` locally and I don't think it produce an error, so maybe even the `rm -rf` isn't necessary.

Reply:
Just a note here if we go the `rm -rf ios/build/WDA` to utilize https://www.npmjs.com/package/rimraf for cross-platform compatibility. If it's not needed here then please do ignore 🙇🏾

Reply:
>When we put test:e2e:build-wda inside test:e2e:ios:local it should also always rebuild WDA when ran locally, right? We can add a rm -rf ios/build/WDA in test:e2e:build-wda to be safe, but it seems to me that it's the same case for building with test:e2e:build-app:ios locally and I don't think it produces an error, so maybe even the rm -rf isn't necessary.

Yeah you're right, we can run the command `test:e2e:build-wda` when running `test:e2e:ios:local`. I thought that the workflow was running `test:e2e:ios:local` but it's not, sorry for the misleading. Since this command is only run locally, it makes sense to also build the WDA. This would also require to pass the WDA derived data folder as an environment variable when running it from Gutenberg-mobile, as we do with the app’s path but nothing else.
 
Regarding removing the derived data folder, I'm going to check it but most likely it won't be necessary.

Thanks for the feedback @ceyhun and @jd-alexander 🙇 !

Reply:
I applied the changes from these comments in this [commit](https://github.com/WordPress/gutenberg/pull/29831/commits/f02b756a056b808e0a5e5509e190fe4aeb45e709). I'm going to open a Gutenberg-mobile PR with the required changes to run the e2e tests there too.

Reply:
Here is the [PR](https://github.com/wordpress-mobile/gutenberg-mobile/pull/3266) in Gutenberg-mobile related to these changes, @ceyhun and @jd-alexander I've assigned you as reviewers there too.